src/argilla/server/models.py

```diff
@@ -34,26 +34,54 @@ def default_inserted_at(context):
     return context.get_current_parameters()["inserted_at"]
 
 
+class AnnotationType(str, Enum):
+    text = "text"
+    single_label = "single_label"
+
+
 class UserRole(str, Enum):
     admin = "admin"
     annotator = "annotator"
 
 
+class Annotation(Base):
+    __tablename__ = "annotations"
+
+    id: Mapped[UUID] = mapped_column(primary_key=True, default=uuid4)
+    name: Mapped[str]
+    title: Mapped[str] = mapped_column(Text)
+    type: Mapped[Optional[AnnotationType]] = mapped_column(default=AnnotationType.text)
+    required: Mapped[Optional[bool]] = mapped_column(default=False)
+    settings: Mapped[Optional[dict]] = mapped_column(JSON, default={})
```

Comment:
Same for these fields
```suggestion
    required: Mapped[bool] = mapped_column(default=False)
    settings: Mapped[dict] = mapped_column(JSON, default={})
```

Reply:
The same applies here.
* `required` has a default value of `False` so it's optional. If we don't specify a value for `required` the annotation then is not required. Of course we can do the opposite, if a value for `required` is not specified by default the annotation is required (using `True` as default).
* I was thinking that maybe we don't have a `settings` value for all kind of annotations so I'm using as default an empty JSON object here (or Python empty dict) that's why is optional too.